yarn-project/archiver/src/archiver/archiver_store_test_suite.ts

```diff
@@ -354,6 +354,110 @@ export function describeArchiverDataStore(testName: string, getStore: () => Arch
       });
     });
 
+    describe('getLogsByTags', () => {
+      const txsPerBlock = 4;
+      const numPrivateFunctionCalls = 3;
+      const numNoteEncryptedLogs = 2;
+      const numBlocks = 10;
+      let blocks: L1Published<L2Block>[];
+      let tags: { [i: number]: { [j: number]: Fr[] } } = {};
+
+      beforeEach(async () => {
+        blocks = times(numBlocks, (index: number) => ({
+          data: L2Block.random(index + 1, txsPerBlock, numPrivateFunctionCalls, 2, numNoteEncryptedLogs, 2),
+          l1: { blockNumber: BigInt(index), blockHash: `0x${index}`, timestamp: BigInt(index) },
+        }));
+        // Last block has the note encrypted log tags copied from the previous block
+        blocks[numBlocks - 1].data.body.noteEncryptedLogs.txLogs.forEach((txLogs, txIndex) => {
+          txLogs.functionLogs.forEach((fnLogs, fnIndex) => {
+            fnLogs.logs.forEach((log, logIndex) => {
+              const previousLogData =
+                blocks[numBlocks - 2].data.body.noteEncryptedLogs.txLogs[txIndex].functionLogs[fnIndex].logs[logIndex]
+                  .data;
+              previousLogData.copy(log.data, 0, 0, 32);
+            });
+          });
+        });
+
+        await store.addBlocks(blocks);
+        await store.addLogs(blocks.map(b => b.data));
```

Comment:
Let's also add a scenario in which we add a log that does not have a tag and have it not fail (though we of course cannot fetch it by tag).

Reply:
Added a tx full of invalid tags just in case